### PR TITLE
Add dynamic card chooses title

### PR DESCRIPTION
**Pull request by tintinthong**
The goal of this PR is have dynamic UI to suggest to the user what type of Card they are choosing within the CARD_CHOOSER

The places that need attention are
- 1. `links-to-editor` 
Screenshot 2023-07-23 at 3 15 12 PM
Screenshot 2023-07-23 at 3 15 22 PM
- 2. `links-to-many-component`
Screenshot 2023-07-23 at 3 11 15 PM
Screenshot 2023-07-23 at 3 11 34 PM
- 3. `file-tree`
Screenshot 2023-07-23 at 3 17 16 PM
Screenshot 2023-07-23 at 3 17 25 PM
- 4. `schema`. This view returns the default as the suggestCardChooserTitle doesn't parse 'not' query 
Screenshot 2023-07-23 at 3 20 58 PM
Screenshot 2023-07-23 at 3 41 35 PM
- 5. `container`
Screenshot 2023-07-23 at 3 22 07 PM
Screenshot 2023-07-23 at 3 22 13 PM
- 6. `room`

### CODE REVIEW

**Review by lukemelia — CHANGES_REQUESTED** [review]
Looks promising. Can we add some tests?

File: packages/runtime-common/text-suggestion.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import { isCardTypeFilter, isEveryFilter, type Filter } from './query';
+
+interface ChooseCardSuggestion {
+  suggestion: string; // suggests a UI text
+  depth: number;
+}
+export function suggestCardChooserTitle(
```
Comment (tintinthong) [inline packages/runtime-common/text-suggestion.ts]: This function tries to make suggestions as to what text when choosing a list of cards based upon the `Filter`. Particularly, by searching for possible cases of card context like `on` or `type`. 
This function uses a recursive call when going into quantifier like `every`. But not all filters types are supported like `any` or `not` or `range`. The reason is bcos these filters dont contribute to context of the card search and possibly confound

I wonder in the future will we have other use cases where we dynamically suggest text based upon the query.

**Review by burieberry — COMMENTED** [review]
Overall looks good, a couple of points:
- Need to use `a` or `an` depending on context: `Choose an Article card`, `Choose a Post card`
- In the LinksToMany case, the title will ask to select 1 or more cards `Select 1 or more contact cards`

**Comment by tintinthong** [comment]
> Overall looks good, a couple of points:
> 
> * Need to use `a` or `an` depending on context: `Choose an Article card`, `Choose a Post card`
> * In the LinksToMany case, the title will ask to select 1 or more cards `Select 1 or more contact cards`

Great suggestions! Additionally I moved text-suggestions into host from runtime common

Screenshot 2023-07-26 at 4 47 27 PM
Screenshot 2023-07-26 at 4 47 44 PM
Screenshot 2023-07-26 at 4 49 08 PM

**Review by FadhlanR — APPROVED** [review]

**Comment by tintinthong** [comment]
It should be "Choose 1 or more Friend Cards" (+s) now

**Review by burieberry — APPROVED** [review]

**Pull request merged**